packages/ui/src/components/common/TransactionModal/steps/SwapConfirmationStep.tsx

```diff
@@ -1,33 +1,163 @@
-import { type FC } from 'react'
+import { useMemo, type FC } from 'react'
 import {
-  Button,
   Flex,
   Text,
-  ChainTokenIcon
+  ChainTokenIcon,
+  Box,
+  Anchor,
+  ChainIcon
 } from '../../../primitives/index.js'
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
-import { LoadingSpinner } from '../../LoadingSpinner.js'
 import { type Token } from '../../../../types/index.js'
 import { faArrowRight } from '@fortawesome/free-solid-svg-icons/faArrowRight'
 import type { useQuote } from '@reservoir0x/relay-kit-hooks'
-import { formatDollar } from '../../../../utils/numbers.js'
+import type { Execute, RelayChain } from '@reservoir0x/relay-sdk'
+import useRelayClient from '../../../../hooks/useRelayClient.js'
+import { faCheck, faExternalLink } from '@fortawesome/free-solid-svg-icons'
+import { getTxBlockExplorerUrl } from '../../../../utils/getTxBlockExplorerUrl.js'
+import { truncateAddress } from '../../../../utils/truncate.js'
+import { getStepActionText } from '../../../../utils/getStepActionText.js'
+
+type FormattedStep = {
+  id: string
+  action: string
+  isActive: boolean
+  isCompleted: boolean
+  progressState?: NonNullable<
+    Execute['steps']['0']['items']
+  >[0]['progressState']
+  txHashes?: { txHash: string; chainId: number }[]
+  isWalletAction: boolean
+  chainId?: number
+  isApproveStep?: boolean
+}
 
 type SwapConfirmationStepProps = {
   fromToken?: Token
   toToken?: Token
-  quote?: ReturnType<typeof useQuote>['data']
+  fromChain?: RelayChain
+  toChain?: RelayChain
   fromAmountFormatted: string
   toAmountFormatted: string
+  quote?: ReturnType<typeof useQuote>['data']
+  steps: Execute['steps'] | null
 }
 
 export const SwapConfirmationStep: FC<SwapConfirmationStepProps> = ({
   fromToken,
   toToken,
-  quote,
+  fromChain,
+  toChain,
   fromAmountFormatted,
-  toAmountFormatted
+  toAmountFormatted,
+  quote,
+  steps
 }) => {
-  const details = quote?.details
+  const operation = quote?.details?.operation || 'swap'
+  const isSameChain = fromChain?.id === toChain?.id
+
+  /**
+   * formattedSteps transforms backend transaction steps into user-friendly UI steps by:
+   * - Tracking active/delayed/completed states
+   * - Adding readable action descriptions
+   * - Extracting transaction hashes for explorer links
+   * - Adding a final Relay processing step
+   */
+  const { formattedSteps, status } = useMemo(() => {
```

Comment:
Should this hook be in it's own file for readability sake?

Reply:
Yeah good idea